components/time-lag-chart/index.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import {
+  Cell,
+  Bar,
+  BarChart,
+  ReferenceLine,
+  XAxis,
+  Tooltip,
+  ResponsiveContainer,
+} from 'recharts'
+
+import data from './data'
+
+const formatter = value => {
+  switch (value) {
+    case -365:
+      return '-1y'
+    case -31:
+      return '-1m'
+    case -7:
+      return '-1w'
+    case 0:
+      return '0'
+    case 7:
+      return '1w'
+    case 31:
+      return '1m'
+    case 90:
+      return '90d'
+    case 365:
+      return '1y'
+    case 365 * 2:
+      return '2y'
+    case 365 * 3:
+      return '3y'
+    default:
+      return `${value}`
+  }
+}
+
+const ticks = [-365, -31, -7, 0, 7, 31, 90, 365]
```

Comment:
I would prefer automatic logarithmic axis. Is it possible? How much effort does it require? Do we need D3 for it?

Reply:
Good idea! Log scale for YAxis shouldn't be that hard to do. Recharts has [API](http://recharts.org/en-US/api/YAxis#scale) for it but it didn't work. I will need to raise an issue. Nonetheless, we can use D3 directly if it doesn't work.

I am not sure about Xaxis. The type of Xaxis is categorical, not numeric. I don't you can apply log scale on this, can you? We would need to develop our own scale function. Let's wait for the backend to see what data we will get. :)